windows/ReactTestApp/MainPage.cpp

```diff
@@ -100,14 +92,12 @@ namespace winrt::ReactTestApp::implementation
     MenuFlyoutItem MainPage::MakeComponentMenuButton(::ReactTestApp::Component &component)
     {
         hstring componentDisplayName = to_hstring(component.displayName.value_or(component.appKey));
-        hstring appKey = to_hstring(component.appKey);
-        ReactTestApp::ComponentViewModel newComponent =
-            winrt::make<ComponentViewModel>(appKey, componentDisplayName);
 
         MenuFlyoutItem newMenuItem;
-        newMenuItem.Text(newComponent.DisplayName());
-        newMenuItem.CommandParameter(newComponent);
-        newMenuItem.Click({this, &MainPage::SetReactComponentName});
+        newMenuItem.Text(componentDisplayName);
+        newMenuItem.Click([this, component](IInspectable const &, RoutedEventArgs) {
```

Comment:
Avoid copying `component`:
```suggestion
        newMenuItem.Click([this, &component](IInspectable const &, RoutedEventArgs) {
```

Reply:
When passing `component` by reference, component.appKey is empty string and component.initialProperties is `nullopt` by the time the code in click handler runs. Could it be that component object is destructed?

Reply:
Did we make sure to pass `component` by reference all the way? If we made a copy somewhere, this would fail.

Reply:
Oh, yes... We should make a copy here since `manifest` will be destructed when `MainPage::MainPage()` returns:
```cpp
    MainPage::MainPage()
    {
        ...
        std::optional<::ReactTestApp::Manifest> manifest = ::ReactTestApp::GetManifest();
        ...
    }
```